pkg/ccl/backupccl/split_and_scatter_processor.go

```diff
@@ -83,14 +81,18 @@ func (s dbSplitAndScatterer) splitAndScatterKey(
 		return 0, nil
 	}
 
-	return s.findDestination(ctx, br), nil
+	return s.findDestination(res.(*roachpb.AdminScatterResponse)), nil
 }
 
 // findDestination returns the node ID of the node of the destination of the
 // AdminScatter request. If the destination cannot be found, 0 is returned.
-func (s dbSplitAndScatterer) findDestination(
-	_ context.Context, _ *roachpb.BatchResponse,
-) roachpb.NodeID {
+func (s dbSplitAndScatterer) findDestination(res *roachpb.AdminScatterResponse) roachpb.NodeID {
+	// A request from a 20.1 node will not have a RangeInfos with a lease.
+	if len(res.RangeInfos) > 0 {
+		// If the lease is not populated, we return the 0 value anyway.
+		return res.RangeInfos[0].Lease.Replica.NodeID
```

Comment:
Might also be nice to explain what >1 rangeInfos means and why only looking at first is OK